### ISSUE TITLE
Support `CONFIG_MODVERSIONS`

### ISSUE THREAD

**Issue by ojeda** [Rust-for-Linux/linux#59]

**Comment by stappersg** [comment]
Adding string `ERROR: modpost: too long symbol`
and [mailinglist Rust-for-Linux archive link](https://lore.kernel.org/rust-for-linux/YK6Vv0fIsSRz8hLR@kroah.com/T/#mc3de4a8b5abbb4b819739f7b4a7b077e9edc0b9a).

The idea is to provide some context to this issue. (And yes, I took the risk of adding noise (instead of context))

**Comment by GuilhermeGiacomoSimoes** [comment]
@ojeda 
We have anybody working on MODVERSIONS support for rust modules ?
We have any discussion about this (zulipchat, mailing list...)  ?

**Comment by ojeda** [comment]
Yeah, please see https://lore.kernel.org/rust-for-linux/20250103-extended-modversions-v13-0-35d87c65ee04@google.com/ and https://lore.kernel.org/rust-for-linux/20250103204521.1885406-20-samitolvanen@google.com/

**Comment by ojeda** [comment]
Since ac61506bf2d1 ("rust: Use gendwarfksyms + extended modversions for CONFIG_MODVERSIONS") we have:

```txt
	select EXTENDED_MODVERSIONS if MODVERSIONS
	depends on !MODVERSIONS || GENDWARFKSYMS
```
